src/wallet/wallet.cpp

```diff
@@ -4401,6 +4401,12 @@ void CWallet::postInitProcess()
     // Add wallet transactions that aren't already in a block to mempool
     // Do this here as mempool requires genesis block to be loaded
     ReacceptWalletTransactions();
+
+    // Update wallet transactions with current mempool transactions.
+    LOCK(::mempool.cs);
```

Comment:
nit, this lock is not at the begin of a scope, could move this to a new `CWallet` member.